src/Compilers/Core/Portable/SourceFileResolver.cs

```diff
@@ -87,14 +152,17 @@ public override bool Equals(object obj)
             }
 
             var other = (SourceFileResolver)obj;
-            return string.Equals(_baseDirectory, other._baseDirectory, StringComparison.Ordinal) &&
-                _searchPaths.SequenceEqual(other._searchPaths, StringComparer.Ordinal);
+            return
+                string.Equals(_baseDirectory, other._baseDirectory, StringComparison.Ordinal) &&
+                _searchPaths.SequenceEqual(other._searchPaths, StringComparer.Ordinal) &&
+                (_pathMap.IsDefaultOrEmpty ? other._pathMap.IsDefaultOrEmpty : _pathMap.SequenceEqual(other._pathMap));
```

Comment:
It looks like there are no checks in the constructor that wold prevent _pathMap from having default value. Consider using NullToEmpty helper.